cirq-google/cirq_google/calibration/phased_fsim.py

```diff
@@ -284,6 +311,34 @@ def get_parameters(self, a: Qid, b: Qid) -> Optional['PhasedFSimCharacterization
         else:
             return None
 
+    # Workaround for: https://github.com/python/mypy/issues/1362
+    @property  # type: ignore
+    @lru_cache_typesafe
```

Comment:
If we remove the frozen attribute and the hashability of this class, you can remove lru_cache and instead cache the `Engine` object on the object the normal way

```python
if self._engine is not None:
  return self._engine
self._engine = ...
```

Reply:
Done